CONTRIBUTING.md

```diff
@@ -75,6 +82,116 @@ $ bundle exec rake compile -- --with-openssl-dir=$HOME/.openssl/openssl-3.0.1
 $ bundle exec rake test
 ```
 
+#### With OpenSSL master branch
+
+If you want to test, debug, report an issue or contribute to the Ruby OpenSSL
+or [the OpenSSL project](https://www.openssl.org/) including both non-FIPS and
+the [FIPS](https://github.com/openssl/openssl/blob/master/README-FIPS.md) cases,
+you can run the following commands to install the OpenSSL. Testing with the
+latest OpenSSL master branch is a good practice to report an issue to the
+OpenSSL project.
+
+The following configuration options are useful in this case.
+
+* `enable-fips`: Add an option to run with the OpenSSL FIPS module.
+* `enable-trace`: Add an option to enabling tracing log. You can trace logs by
+  implementing a code. See
+  [OSSL_TRACE](https://www.openssl.org/docs/man3.0/man3/OSSL_TRACE.html) for details.
+* compiler flags: You can set debugging compiler flags.
+
+```
+$ git clone https://github.com/openssl/openssl.git
+$ git branch | grep '^*'
+* master
+$ cd openssl
+$ git rev-parse --short HEAD
+0bf18140f4
+$ ./Configure \
+  --prefix=$HOME/.openssl/openssl-fips-debug-0bf18140f4 \
+  --libdir=lib \
+  shared \
+  enable-fips \
+  enable-trace \
+  -O0 -g3 -ggdb3 -gdwarf-5
```

Comment:
We can pass `'-Wl,-rpath,$(LIBRPATH)'` to OpenSSL's `Configure` and `LD_LIBRARY_PATH=` will become unnecessary. This is suggested in OpenSSL's docs: https://github.com/openssl/openssl/blob/c7ed5e4697a71012e0a2d9dd5eaf997754ae5156/NOTES-UNIX.md

Actually, the current example (on line 68 in master) is using neither rpath nor LD_LIBRARY_PATH so it most likely wouldn't work as intended... Can you fix it too while you're at it?

Reply:
Oh, you are right! I was not familiar with the way to compile with the `-rpath` option yet. I will fix it with the configure option. It is definitely a better way when we don't need to set the `LD_LIBRARY_PATH` every time.

And OK. I will also fix the current example too.

Reply:
I fixed it.